packages/cli-kit/src/bugsnag.ts

```diff
@@ -0,0 +1,52 @@
+import * as path from './path.js'
+import {Interfaces} from '@oclif/core'
+import Bugsnag from '@bugsnag/js'
+import {realpath} from 'fs/promises'
+
+/**
+ * If the given file path comes from within a plugin, return the relative path, plus the plugin name.
+ *
+ * This gives us very consistent paths for errors thrown from plugin code.
+ *
+ */
+export function cleanStackFrameFilePath(
+  currentFilePath: string,
+  projectRoot: string,
+  pluginLocations: {name: string; pluginPath: string}[],
+): string {
+  const fullLocation = path.isAbsolute(currentFilePath) ? currentFilePath : path.join(projectRoot, currentFilePath)
+
+  const matchingPluginPath = pluginLocations.filter(({pluginPath}) => fullLocation.indexOf(pluginPath) === 0)[0]
+
+  if (matchingPluginPath !== undefined) {
+    return path.join(matchingPluginPath.name, path.relative(matchingPluginPath.pluginPath, fullLocation))
+  }
+  return currentFilePath
```

Comment:
It's hard to tell what path manipulations are happening here. What about adding an example of an input and an output?